docs/src/broadcast_background.md

```diff
@@ -1,11 +1,97 @@
 # Julia Base Broadcast Background
 
-Please see the official Julia [broadcasting documentation]
+Here, we'll provide our own brief background of Julia Base's broadcast
+machinery. Please see the official Julia [broadcasting documentation]
 (https://docs.julialang.org/en/v1/manual/arrays/#Broadcasting), for more
 information.
 
-Here, we'll give a brief background of Julia Base's broadcast machinery.
+## Introduction
 
-## From parse time to runtime
+Sometimes, you need to perform operations on arrays of different sizes, like
+adding a vector to each column of a matrix. One inefficient approach would be
+to expand the vector to match the matrix size:
+
+```julia
+julia> a = rand(2, 1); A = rand(2, 3);
+
+julia> repeat(a, 1, 3) + A
+2×3 Array{Float64,2}:
+ 1.20813  1.82068  1.25387
+ 1.56851  1.86401  1.67846
+```
+
+This can be inefficient with large arrays, so Julia offers the broadcast
+function. It automatically expands smaller dimensions in the arrays to match
+the larger ones without using extra memory and applies the operation element by
+element:
+
+```julia
+julia> broadcast(+, a, A)
+2×3 Array{Float64,2}:
+ 1.20813  1.82068  1.25387
+ 1.56851  1.86401  1.67846
+
+julia> b = rand(1,2)
+1×2 Array{Float64,2}:
+ 0.867535  0.00457906
+
+julia> broadcast(+, a, b)
+2×2 Array{Float64,2}:
+ 1.71056  0.847604
+ 1.73659  0.873631
+```
+
+Operators like `.+` and `.*` are shorthand for broadcast calls (with the added
+benefit of "fusing" the operations). The `broadcast!` function lets you specify
+where to store the result (and you can use `.=` to do this in a similar fused
+way). For example, `f.(args...)` is the same as `broadcast(f, args...)`,
+offering a simpler syntax for broadcasting any function. Also, nested "dot
+calls" like `f.(...)` automatically fuse into a single broadcast call.
```

Comment:
```
Also, nested "dot calls" like `f.(...)` automatically fuse into a single broadcast call.
```

I find this phrasing a little unclear

Reply:
Ah, yeah, let me improve that, I think I know how.

Reply:
Updated!